sstables/mx/writer.cc

```diff
@@ -608,6 +608,7 @@ class writer : public sstable_writer::writer_impl {
     } _pi_write_m;
     utils::UUID _run_identifier;
     bool _write_regular_as_static; // See #4139
+    bool _cannot_have_partition_tombstones = true;
 
```

Comment:
Maybe we should collect the lowest partition tombstone timestamp instead (and if there are none, it's the maximum timestamp). Maybe this can help with something, and has exactly the same collection cost.

@tgrabiec @raphaelsc maybe you can find a use for it.

Reply:
btw, the reason I am expanding the scope is that I'd like to make format changes as rare as we can.

Reply:
in sst 3 format, there's min deletion time metadata to track if a sstable has tombstones of any sort, so it makes sense to have min partition tombstone deletion time to track for existence of partition tombstone.

Reply:
> min deletion time metadata to track if a sstable has tombstones of any sort

Note that it's also updated on TTLs, hence it cannot be really used to track "real" tombstones. That's why I needed this new metadata.

Reply:
> it makes sense to have min partition tombstone deletion time to track for existence of partition tombstone

But would it have an actual use? I could modify the PR to collect a timestamp instead of a flag but honestly I like it the way it is now :)

Reply:
Btw. if we want to avoid format changes, maybe we could somehow make an extensible "statistics" metadata entry. Like `stats_metadata` but Scylla specific. However, `stats_metadata` is not extensible as far as I understand, and I don't know how we could make a nice extensible field (which can be appended to over time). I guess we could use a `disk_array` but that wouldn't be very elegant since it doesn't have type information. It would be best to represent the field as a `struct`. It probably could be done with a sufficient amount of template magic but I don't know if doing this as part of this PR wouldn't miss the point (I just want to finish #7437 :( ).

Reply:
with the boolean, we gain simplicity and I like it more too. with the min timestamp, we have extra information that may or may not be required later on. however, extra information is useless if it will never be used. so we'd only go with the latter if anybody can think of where it could possibly be applied in the future.

Reply:
> with the boolean, we gain simplicity and I like it more too. with the min timestamp, we have extra information that may or may not be required later on. however, extra information is useless if it will never be used. so we'd only go with the latter if anybody can think of where it could possibly be applied in the future.

also it's not the end of the world if we eventually end up adding a new field. It's inside Scylla-specific metadata anyway which works in a backwards-compatible way as far as I understand (does it?).

Reply:
> > with the boolean, we gain simplicity and I like it more too. with the min timestamp, we have extra information that may or may not be required later on. however, extra information is useless if it will never be used. so we'd only go with the latter if anybody can think of where it could possibly be applied in the future.
> 
> also it's not the end of the world if we eventually end up adding a new field. It's inside Scylla-specific metadata anyway which works in a backwards-compatible way as far as I understand (does it?).

it does. scylla_metadata is capable of determining whether a metadata is available or not, so it's important to have a safe default value, like you did.

Reply:
but please keep in mind that it's important to anticipate future usage of metadata to prevent scylla component from being eventually bloated. if we know that adding a bit more info to a metadata field can *very likely* be useful down the road, then go for it.